### DISCUSSION TITLE
Mainnet contract cannot be verified

### DISCUSSION THREAD

**Discussion by machaohappyii** [zkSync-Community-Hub/zksync-developers#379] [Contract Verification]
### Environment

Mainnet

### Issue Type

- [X] Plugin
- [X] Block Explorer

### Contract Address

0x59F08deF0F1920CC41C022f03BF4d3B6A6B09F5D

### Compiler Type

Single file

### zkSolc Version

v1.4.0

### Solc Version

0.8.17

### Contract Name

GreeterXiaoma 

### Contract Code

//SPDX-License-Identifier: Unlicense
pragma solidity ^0.8.17;

contract GreeterXiaoma {
    string private greeting;

    constructor(string memory _greeting) {
        greeting = _greeting;
    }

    function greet() public view returns (string memory) {
        return greeting;
    }

    function setGreeting(string memory _greeting) public {
        greeting = _greeting;
    }
}

### Constructor Arguments

0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000094869207468657265210000000000000000000000000000000000000000000000

### Additional Details

![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/7270823/e882cf22-4d7f-4f8b-889d-8e3f7e7451fb)

**Comment by dutterbutter** [comment]
Hey @machaohappyii can you share the code you are using to verify and the hardhat config please.

**Comment by npam90** [comment]
The provided Solidity contract "GreeterXiaoma" has been deployed to the mainnet at address 0x59F08deF0F1920CC41C022f03BF4d3B6A6B09F5D.

**Comment by dutterbutter** [comment]
@machaohappyii is this still an issue?

**Comment by deosovitis** [comment]
The user is requesting assistance with verifying a Solidity contract on the Mainnet with zkSolc v1.4.0 and Solc v0.8.17. The contract code is for a simple Greeter contract with a constructor and two functions. The constructor arguments are provided in hexadecimal format. No repository link or information about the Hardhat Verify plugin version was provided. A screenshot of the contract code was included.
